src/main/java/com/pinterest/secor/uploader/Uploader.java

```diff
@@ -67,21 +67,40 @@ public Uploader(SecorConfig config, OffsetTracker offsetTracker, FileRegistry fi
     }
 
     private Future<?> upload(LogFilePath localPath) throws Exception {
-        String s3Prefix = "s3n://" + mConfig.getS3Bucket() + "/" + mConfig.getS3Path();
-        LogFilePath s3Path = new LogFilePath(s3Prefix, localPath.getTopic(),
+    	String prefix = "";
+        if (mConfig.getCloudService() != null && mConfig.getCloudService().equals("Swift")){ // the first condition 
```

Comment:
Could you move this code to a separate method called getPrefix or something similar?